### DISCUSSION TITLE
Question about why certain data is queried in getStaticProps in api-routes-apollo-server-and-client example

### DISCUSSION THREAD

**Discussion by hyunwoona** [vercel/next.js#27852] [Help]
Hi guys, I'm new to both Next.js and Apollo, so I'm still figuring things out, but I was looking at `api-routes-apollo-server-and-client` example for create-next-app, and really don't understand something.

in https://github.com/vercel/next.js/blob/canary/examples/api-routes-apollo-server-and-client/pages/index.js
```
export async function getStaticProps() {
  const apolloClient = initializeApollo()

  await apolloClient.query({
    query: ViewerQuery,
  })

  return {
    props: {
      initialApolloState: apolloClient.cache.extract(),
    },
  }
}
```

it's querying and caching `viewer` here.

`viewer` in this example is basically the user that's using the app.

From what I've seen so far,  in `getStaticProps`, they query something that doesn't really change, and something that's less specific, like all countries, or all teams, etc. since you can't pass the ID for a viewer? (in the example it just returns a hardcoded object with ID of 1) and if you want to pass parameters, you would use a dynamic route like `[id].js`?

In this example, I'm not sure how useful querying and caching the `viewer` will be... or how to make it work when there are multiple users...
so in short, the question is what kind of data should be queried and cached from `getStaticProps` and why querying and caching `viewer` here is a good decision? and how do I make it work with multiple users in the table?
